src/app/plugin/pluginManager.js

```diff
@@ -1,13 +1,47 @@
 'use strict'
 /**
  * Register and Manage plugin:
+ *
  * Plugin registration is done in the settings tab,
  * using the following format:
  * {
  *  "title": "<plugin name>",
  *  "url": "<plugin url>"
  * }
  *
+ * structure of messages:
+ *
+ * - Notification sent by Remix:
+ *{
+ *  action: 'notification',
+ *  key: <string>,
+ *  type: <string>,
+ *  value: <array>
+ *}
+ *
+ * - Request sent by the plugin:
+ *{
+ *  id: <number>,
+ *  action: 'request',
+ *  key: <string>,
+ *  type: <string>,
+ *  value: <array>
+ *}
+ *
+ * - Response sent by Remix and receive by the plugin:
+ *{
+ *  id: <number>,
+ *  action: 'response',
+ *  key: <string>,
+ *  type: <string>,
+ *  value: <array>,
+ *  error: (see below)
+ *}
```

Comment:
I just checked again: https://github.com/ethereum/remix-ide/issues/1156

* The original proposal is `var { id, status, type, key, value } = msg`
* You said you would like to add the `action` field to indicate "response/notification/request".
* That would make it `var { action, id, status, type, key, value } = msg`
* But now the `status` is/was renamed to `error` i guess.

Just to make sure how i have to think about it :-)
So i guess we now have: `var { action, id, error: status, type, key, value } = msg`

I would say that is fine.

Reply:
The above comment is just for me to clarify.
If this is how it's meant, i would say the PR looks good to me :-)

Reply:
yes except that error is undefined if there's no error